lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -399,33 +430,30 @@ def _needs_upgrade(self, plugin: OpenSearchPlugin) -> bool:
         num_points = min(len(plugin_version), len(version))
         return version[:num_points] != plugin_version[:num_points]
 
-    def _is_plugin_relation_set(self, relation_name: str) -> bool:
-        """Returns True if a relation is expected and it is set."""
-        if not relation_name:
-            return False
-        relation = self._charm.model.get_relation(relation_name)
-        if self._event_scope == OpenSearchPluginEventScope.RELATION_BROKEN_EVENT:
-            return relation is not None and relation.units
-        return relation is not None
-
     def _remove_if_needed(self, plugin: OpenSearchPlugin) -> bool:
         """If disabled, removes plugin configuration or sets it to other values."""
-        if self.status(plugin) == PluginState.DISABLED:
-            if plugin.REMOVE_ON_DISABLE:
-                return self._remove_plugin(plugin)
         return False
 
     def _remove_plugin(self, plugin: OpenSearchPlugin) -> bool:
         """Remove a plugin without restarting the node."""
         try:
             self._opensearch.run_bin("opensearch-plugin", f"remove {plugin.name}")
+            self._clean_cache_if_needed()
+
         except OpenSearchCmdError as e:
             if "not found" in str(e):
                 logger.info(f"Plugin {plugin.name} to be deleted, not found. Continuing...")
                 return False
             raise OpenSearchPluginRemoveError(plugin.name)
         return True
 
+    def _clean_cache_if_needed(self):
```

Comment:
**nit**: Not sure why this is needed. Clearing the cache on `self.plugins` will lead to `ConfigExposedPlugins` being read again, which is static, so no benefit to clearing. `self.plugins` is being used as a `read_only` property as far as I can see. 

For `_installed_plugins` I would make that a `@property` instead, and just evaluate every time.